src/lifecycle.mjs

```diff
@@ -182,43 +186,41 @@ export async function init(worker, crawlPath) {
     )}`
   );
 
-  for await (const segment of crawlPath) {
-    await Promise.all(
```

Comment:
@il3ven I removed Promise.all here as effectively I don't think segment is really returning an array of promises. From what I understand: For that, within `async (strategy) =>` we'd have to return.

However, I think my solution is equivalent in terms of functionality but a bit more readable.